core/cc/linux/debugger.cpp

```diff
@@ -20,6 +20,10 @@ static volatile bool gIsDebuggerAttached = false;
 
 namespace core {
 
+void Debugger::waitForAttach() {
+    while (!isAttached()) {}
+}
+
 bool Debugger::isAttached() {
```

Comment:
It seems a bit redundant.
But the real reason is that if it is removed, all that is needed is to 1) attach debugger 2) force return from function.
With the extra function and without symbols, that attach flow no longer works.